nx_neptune/instance_management.py

```diff
@@ -569,6 +573,73 @@ def stop_na_instance(
         return _invalid_status_code(status_code, response)
 
 
+def update_na_instance_size(
+    graph_id: str,
+    prospect_size=None,
+    sts_client: Optional[BaseClient] = None,
+    iam_client: Optional[BaseClient] = None,
+    na_client: Optional[BaseClient] = None,
+):
+    """Update the provisioned memory size of a Neptune Analytics graph instance.
+
+    This function handles updating the memory size of an existing Neptune Analytics graph instance.
+    If no prospect_size is provided, it will double the current size. The new size must not exceed
+    the limit specified in NETWORKX_GRAPH_SIZE_LIMIT environment variable.
+
+    Args:
+        graph_id (str): The ID of the Neptune Analytics graph to resize
+        prospect_size (int, optional): The desired new memory size in GB. If None, doubles current size.
+        sts_client (Optional[BaseClient]): Optional STS boto3 client. If not provided, a new one will be created.
+        iam_client (Optional[BaseClient]): Optional IAM boto3 client. If not provided, a new one will be created.
+        na_client (Optional[BaseClient]): Optional Neptune Analytics boto3 client. If not provided, a new one will be created.
+
+    Returns:
+        asyncio.Future: A Future that resolves when the resize operation completes
+
+    Raises:
+        Exception: If the resize operation fails or exceeds size limits
+        ValueError: If the role lacks required permissions or NETWORKX_GRAPH_SIZE_LIMIT is not set
+    """
+    (iam_client, na_client) = _get_or_create_clients(sts_client, iam_client, na_client)
+    # Permission check
+    iam_client.has_update_na_permissions()
+
+    raw_size_limit = os.getenv(_ENV_SIZE_LIMIT)
```

Comment:
why is this an environment variable?  can we default this to the maximum ncu size?